### PR TITLE
scout/advise: first iteration of the `src scout advise` subcommand

### PR DESCRIPTION
**Pull request by jasonhawkharris**
Refactor
- create a universal config object for the scout tool
- before, each subcommand had its own config type
- add `kube` package for repetitive kubernetes actions across sub commands

First iteration of `scout advise`
- scout advise gives recommendations for resources allocations based on current usage

### Test plan
Ships with unit tests. Manually tested on healthy and unhealthy instances of sourcegraph (k8s and Docker).

### CODE REVIEW

**Comment by jasonhawkharris** [comment]
`src scout advise --namespace ns-sourcegraph --pod gitserver-0
https://github.com/sourcegraph/src-cli/assets/62355966/a3875a98-d856-454e-9919-ae7d28d2555c

**Review by jdpleiness — COMMENTED** [review]
I am having some issues trying this out with a local Helm deployment running in Kind. Am I missing something? 
Screenshot 2023-05-30 at 10 42 29 AM

**Comment by jasonhawkharris** [comment]
I have tested this on 
✅ Minikube
✅ GCP GKE
✅ AWS AMI
❌ Kind

I believe that something is happening with Kind, but I have doubts about any of our customers that would use scout would need it to work in a k8s testing environment like Kind. Since it is working in the other deployments, I think this should be merged.

File: cmd/src/scout_advise.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+func init() {
+	cmdUsage := `'src scout advise' is a tool that makes resource allocation recommendations. Based on current usage.
+    Part of the EXPERIMENTAL "src scout" tool.
+    
+    Examples
+        Make recommendations for all pods in a kubernetes deployment of Sourcegraph.
+        $ src scout advise
+        
+        Make recommendations for all containers in a Docker deployment of Sourcegraph.
+        $ src scout advise
```
[the remark this answers is not in the document]
Comment (jasonhawkharris) [reply cmd/src/scout_advise.go:27]: Not yet. This will be in the next PR.

**Pull request merged**